### DISCUSSION TITLE
Feature requests  in Issues or in Discussions?

### DISCUSSION THREAD

**Discussion by stefonarch** [lxqt/lxqt#1881] [Q&A]
Should feature requests go via issues or discussions?

**Comment by tsujan** [comment]
Feature requests should be opened as issues because codes could be written for them.  A report that can be closed by a PR belongs to issues. It can be either a bug report or a feature request.

Unclear features can be discussed in "Discussions" first (as "Ideas" or "Q&A").

**Reply by stefonarch** [reply]
+1, but they should go in Ideas.

**Reply by tsujan** [reply]
I think there's a difference between a practical feature request and a general idea.

For example, there was a user who wanted to make LXQt's appearance like that of macOS. That's an idea not a feature request.

Another user found pcmanfm-qt's item tooltips interfering with his work and wanted them to be optional. This one was a feature request and was implemented.

**Reply by yan12125** [reply]
> For example, there was a user who wanted to make LXQt's appearance like that of macOS. That's an idea not a feature request.

> Another user found pcmanfm-qt's item tooltips interfering with his work and wanted them to be optional. This one was a feature request and was implemented.

Sounds a good criteria. We can turn general or vague issues into discussion threads.

**Comment by scsijon** [comment]
?Is there any reason that you couldn't have a number of sub-menus under Discussions covering each area with generic titles and when it comes to the actual code required, a linked Issue becomes active? It would mean that the argie-bargie discussion could take place outside of the actual code construction/modification area.

It also means that items such as 'cleanup wiki' wouldn't have to appear in issues as it could be covered in a discussions submenu (maybe called something like housekeeping).

just a thought.

**Reply by stefonarch** [reply]
I sure like "Housekeeping"  :)

We need a good structure here, it's day 3 so thingss will evolve. It looks like here a "Feature Requests" section is needed, to collect first ideas, accepting them would lead to an open issue or better PR to discute technical stuff, testing results and so.

Maybe just rename "Ideas" > "Feature Requests?

**Reply by tsujan** [reply]
> It looks like here a "Feature Requests" section is needed

IMHO, the more sections are added, the less practical they'll become, to say nothing of the resulting untidiness.

**Reply by ito32bit** [reply]
> Maybe just rename "Ideas" > "Feature Requests?

"Ideas" > "Ideas and Feature Requests" ?

**Reply by stefonarch** [reply]
> IMHO, the more sections are added, the less practical they'll become, to say nothing of the resulting untidiness.

True, but true is  also the opposite: too less sections and everyhing contains every sort of topic.

There should be few but clear. I'd start with @ito32bit purpose, then we will see.
https://github.com/lxqt/lxqt/discussions/1881#discussioncomment-199148 purposal looks good to me too except for Internationalisation maybe.

**Reply by tsujan** [reply]
> I'd start with @ito32bit purpose

I think it's too soon for any change. We haven't encountered a real problem here ;) We're just speculating. A month or so is needed, at least.

**Comment by scsijon** [comment]
If you keep it to being under Discussions, I suggest you keep it to a 
limit of six or so sub-menu items; 1- Announcements, 2- Requests 
(features and optional external package links), 3- Feedback (general 
questions and the like), 4- International (translations and 
charactersets (always a bane when internationalizations occur)), 5- 
Housekeeping, 6- Related (items not directly related to lxqt, but 
definately worth knowing about, like associated package depreciations or 
major stepups coming soon requiring lxqt changes to be considered), (and 
maybe a) 7- Other, come to mind as a starting set. All required to be 
related to the lxqt github project of course, otherwise it becomes too 
messy to maintain and control. You will need a couple of moderators to 
maintain control (daily visits by at least one and direct contact @mail 
between them) and also to move things around when they appear in the 
wrong area.

If you barred members uploading whole packages and kept it to messages 
and patches only it shouldn't become too 'messy and uncontrollable'. A 
good forum would of course negate some of this, but then you need 
moderators that can dedicate time to control and maintain them across a 
number of years, and that's not a simple task.

Anyway that's one option that came to mind, others are expedential from 
this.

I also like the requirement to use github's login system, as it's of a 
more than robust status than a lot of the others have.

I can't promise a lot of time, but i'll help when able, until I cause 
heartache. Someone will have to teach me the practical end of using 
github though as i'm more a svn guy myself.
